### PR TITLE
Remove Legacy classes and move input functionality

### PR DESCRIPTION
**Pull request by white238**
* Remove legacy classes
* Converts driver to functional classes but doesn't currently work
* Input code was moved but doesn't match up to new classes. This will be piece meal fixed in following PRs.
* Turn off Integration tests until input files are back

### CODE REVIEW

File: .gitlab/build_ruby.yml [hunk 1]
```diff
@@ -40,7 +40,8 @@ ruby-clang_10_0_0-src:
   variables:
     COMPILER: "clang@10.0.0"
     HOST_CONFIG: "ruby-toss_3_x86_64_ib-${COMPILER}.cmake"
-    DO_INTEGRATION_TESTS: "yes"
+    # TODO: Turn this  back on when input files are flushed out
```
Comment (jamiebramwell) [inline .gitlab/build_ruby.yml]:
```suggestion
    # TODO: Turn this  back on when input files are working for the Functional-based physics modules
```

File: examples/simple_conduction/without_input_file.cpp [hunk 2]
```diff
@@ -38,35 +39,40 @@ int main(int argc, char* argv[])
   // _create_mesh_end
 
   // _create_module_start
-  constexpr int order = 2;
-  serac::ThermalConductionLegacy conduction(order, serac::ThermalConductionLegacy::defaultQuasistaticOptions());
+  // Create a Heat Transfer class instance with Order 2 and Dimensions of 3
```
Comment (jamiebramwell) [inline examples/simple_conduction/without_input_file.cpp]:
```suggestion
  // Create a Heat Transfer class instance with Order 1 and Dimensions of 2
```

File: src/docs/sphinx/user_guide/simple_conduction_tutorial.rst [hunk 3]
```diff
@@ -67,144 +64,42 @@ the mesh and fields, in a single ``StateManager`` object, which is initialized h
...
-.. note::
-  The C++-only API requires method calls for configuration, while the Lua-based approach typically only requires changes
-  to the input file.  This is because the configuration options are all part of the ``ThermalConduction::InputOptions``
-  extracted from Inlet.
-
 Configuring Material Conductivity
 ---------------------------------
 
-Instead of using a monolithic material model, the ``ThermalConduction`` module currently allows for material parameters
+Instead of using a monolithic material model, the ``HeatTransfer`` module currently allows for material parameters
```
Comment (jamiebramwell) [inline src/docs/sphinx/user_guide/simple_conduction_tutorial.rst]: This needs to be updated. I will push a fix...

File: src/drivers/CMakeLists.txt [hunk 4]
```diff
@@ -6,28 +6,30 @@
...
-
-    blt_add_test(NAME          serac_driver_no_thermal
-                 COMMAND       ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/serac -o default_no_thermal -i ${input_files_dir}/tests/solid/qs_linear.lua
-                 NUM_MPI_TASKS 1 )
-
-    blt_add_test(NAME          serac_driver_no_solid
-                 COMMAND       ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/serac -o default_no_solid -i ${input_files_dir}/tests/thermal_conduction/static_solve.lua
-                 NUM_MPI_TASKS 1 )
+    # Make sure running serac driver doesn't completely fail
+#TODO: Turn these back on when input files work again
```
Comment (jamiebramwell) [inline src/drivers/CMakeLists.txt]:
```suggestion
#TODO: Turn these back on when input files work with the new physics modules
```

**Comment by white238** [comment]
This is in a state that is now ready to be reviewed. It is mostly deleting code and moving the input into separate files but that will have to be reworked following this PR.

File: src/serac/physics/heat_transfer_input.cpp [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+  // material parameters
+  container.addDouble("kappa", "Thermal conductivity").defaultValue(0.5);
+  container.addDouble("rho", "Density").defaultValue(1.0);
+  container.addDouble("cp", "Specific heat capacity").defaultValue(1.0);
+
+  auto& source = container.addStruct("source", "Scalar source term (RHS of the thermal conduction PDE)");
+  serac::input::CoefficientInputOptions::defineInputFileSchema(source);
+
+  auto& reaction_container = container.addStruct("nonlinear_reaction", "Nonlinear reaction term parameters");
```
Comment (jamiebramwell) [inline src/serac/physics/heat_transfer_input.cpp]: This stuff is no longer needed. I will go in and remove it.

**Review by jamiebramwell — APPROVED** [review]

File: src/serac/physics/solid_mechanics_input.cpp [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "serac/physics/solid_mechanics_input.hpp"
+
+namespace serac {
+
+void SolidMechanicsInputOptions::defineInputFileSchema(axom::inlet::Container& container)
+{
+  // Polynomial interpolation order - currently up to 8th order is allowed
+  container.addInt("order", "Order degree of the finite elements.").defaultValue(1).range(1, 8);
```
Comment (samuelpmishLLNL) [inline src/serac/physics/solid_mechanics_input.cpp]:
```suggestion
  // interpolation order - currently up to 3rd order is allowed
  container.addInt("order", "polynomial order of the basis functions.").defaultValue(1).range(1, 3);
```

**Review by samuelpmishLLNL — APPROVED** [review]

**Pull request merged**